src/framed.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the MIT License, <LICENSE or http://opensource.org/licenses/MIT>.
+// This file may not be copied, modified, or distributed except according to those terms.
+
+use bincode::{SizeLimit, serde as bincode};
+use byteorder::BigEndian;
+use bytes::{Buf, MutBuf};
+use futures::{Async, Poll};
+use serde;
+use std::io::{self, Cursor};
+use std::marker::PhantomData;
+use std::mem;
+use tokio_core::easy::{self, EasyBuf, EasyFramed};
+use tokio_core::io::{FramedIo, Io};
+use tokio_proto::multiplex::{self, RequestId};
+use util::Never;
+
+/// Handles the IO of tarpc messages.
```

Comment:
I guess the main difference is that EasyFramed's type parameters specify the parser and serializer to use -- and by those types, implicitly specifies the types being parsed and serialized -- whereas tarpc::Framed's type parameters are the actual types being parsed and serialized. Not sure if it's worth explaining that in a comment?